packages/flutter/lib/src/services/sensitive_content.dart

```diff
@@ -0,0 +1,156 @@
+  /// projection session.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_SENSITIVE.
+  sensitive,
+
+  /// The widget tree does not contain sensitive content.
+  ///
+  /// When this level is set via a `SensitiveContent` widget, the window
+  /// hosting the screen will only be marked as sensitive if other `SensitiveContent`
+  /// widgets in the Flutter app with the [sensitive] level are present in the widget tree.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_NOT_SENSITIVE.
+  notSensitive,
+
+  /// The sensitivity content level is unknown to Flutter.
+  ///
+  /// This mode may represent the current content sensitivity of the window if, for example, Android
+  /// adds a new mode that is not recognized by the `SensitiveContent` widget.
+  ///
+  /// This mode cannot be used to set the sensitivity level of a `SensitiveContent` widget.
+  _unknown,
+}
+
+/// Service for setting the content sensitivity of the native app window (Android `View`)
+/// that contains the app's widget tree.
+///
+/// This service is only currently supported on Android API 35+.
+class SensitiveContentService {
+  /// Creates service to set content sensitivity of an app window (Android `View`) via
+  /// communication over the sensitive content [MethodChannel].
+  SensitiveContentService() {
+    sensitiveContentChannel = SystemChannels.sensitiveContent;
+  }
+
+  /// The channel used to communicate with the shell side to get and set the
+  /// content sensitivity of an app window (Android `View`).
+  late MethodChannel sensitiveContentChannel;
+
+  /// Sets content sensitivity level of the app window (Android `View`) that contains the app's widget
+  /// tree to the level specified by [contentSensitivity] via a call to the native embedder.
+  void setContentSensitivity(ContentSensitivity contentSensitivity) {
+    try {
+      sensitiveContentChannel.invokeMethod<void>(
+        'SensitiveContent.setContentSensitivity',
+        contentSensitivity.name,
+      );
+    } catch (e) {
+      // Content sensitivity failed to be set.
+      throw FlutterError('Content sensitivity $contentSensitivity failed to be set: $e');
+    }
+  }
+
+  /// Gets content sensitivity level of the app window (Android `View`) that contains
+  /// the app's widget tree.
+  Future<ContentSensitivity> getContentSensitivity() async {
+    String? result;
+    try {
+      result = await sensitiveContentChannel.invokeMethod<String>(
+        'SensitiveContent.getContentSensitivity',
+      );
+    } catch (e) {
+      // Content sensitivity failed to be retrieved.
+      throw FlutterError('Failed to retrieve content sensitivity: $e');
+    }
+
+    final ContentSensitivity contentSensitivity = ContentSensitivity.values.firstWhere(
+      (ContentSensitivity cs) => cs.name == result,
+    );
+
+    if (contentSensitivity == ContentSensitivity._unknown) {
+      throw UnsupportedError(
```

Comment:
Nit: Could you get rid of `ContentSensitivity._unknown` if you throw for any value that doesn't match an enum value? Then you could also get rid of the `ignore_no_default_cases` in widgets/sensitive_content.dart.

```dart
    final ContentSensitivity contentSensitivity = ContentSensitivity.values.firstWhere(
      (ContentSensitivity cs) => cs.name == result,
      orElse: () => throw UnsupportedError(
        'Android Flutter View has a content sensitivity mode '
        'that is not recognized by Flutter. If you see this error, '
        'it is possible the View uses a new mode that Flutter needs to '
        'support; please file an issue.',
      ),
    );

Reply:
Can you remove _unknown from the enum entirely?

Reply:
True, I'm on board with keeping it 👍